app/forms/csv_upload_form.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+class CSVUploadForm
+  include ActiveModel::Validations
+
+  attr_reader :file
+
+  validates_with CSVValidator
+
+  def initialize(user)
+    @user = user
+  end
+
+  def save(params)
+    @file = params[:csv_upload_form][:file]
+
+    return false unless valid?
+
+    begin
+      Keyword.transaction do
+        # rubocop:disable Rails/SkipsModelValidations
+        user.keywords.insert_all parse_keywords
+        # rubocop:enable Rails/SkipsModelValidations
+      end
+    rescue ActiveRecord::StatementInvalid
+      errors.add(:keyword, I18n.t('csv.validation.bad_keyword_length'))
+    end
+
+    errors.empty?
+  end
+
+  private
+
+  attr_reader :user
+
+  def parse_keywords
```

Comment:
Can the function name can be changed to `parsed_keywords`, as we are depending on its return value.

Reply:
Thanks! I'll be more cautious about naming based on returned value :)